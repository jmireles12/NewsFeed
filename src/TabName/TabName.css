.TabName {
    display: inline-flex;
    margin-top: 30px;
    margin-bottom: 30px;
    text-align: center;
}

.tab-name {
    margin: 0;
    padding-right: 10px;
}

.TabName__delete {
    height: fit-content;
    border-radius: 5px;
    border-width: thin;
}