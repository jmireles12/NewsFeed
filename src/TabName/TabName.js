import React, { Component } from 'react'
import ApiContext from '../ApiContext'
import config from '../config'
import './TabName.css'


export default class TabName extends Component {

    static defaultProps = {
        handleDeleteTab: () => { },
        onDeleteTab: () => { },
        match: {
            params: {}
        }
    }

    static contextType = ApiContext

    handleDeleteTab = (e) => {
        const tabId = this.props.id
        fetch(`${config.API_ENDPOINT}/tabs/` + tabId, {
            method: 'DELETE',
            header: {
                'content-type': 'application/json'
            },
            body: JSON.stringify(tabId)
        })
            .then(res => {
                if(!res.ok) {
                    throw new Error(res.statusText)
                }
            })
            .then(res => {
                this.context.deleteTab(tabId)
            })
            .catch(error => {
                console.error(error)
            })
    }

    render() {
        const { name, id } = this.props
        return (
            <section className='TabName'>
                <h4 className='tab-name'>{name}</h4>
                <button
                    className='TabName__delete'
                    type='button'
                    onClick={() => {this.handleDeleteTab(id) }}
                >
                    x
                </button>
            </section>
        )
    }
}