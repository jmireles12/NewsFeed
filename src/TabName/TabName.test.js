import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import TabName from './TabName'

describe(`TabName component`, () => {
    it('renders the complete TabName', () => {
        const wrapper = shallow(<TabName />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})