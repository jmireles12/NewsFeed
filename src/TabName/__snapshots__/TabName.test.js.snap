// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabName component renders the complete TabName 1`] = `
<section
  className="TabName"
>
  <h4
    className="tab-name"
  />
  <button
    className="TabName__delete"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
</section>
`;
